packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts

```diff
@@ -172,46 +172,46 @@ export class OdspDocumentServiceFactoryCore implements IDocumentServiceFactory {
             cacheAndTracker.epochTracker,
         );
     }
+}
 
-    private toInstrumentedOdspTokenFetcher(
-        logger: ITelemetryLogger,
-        resolvedUrl: IOdspResolvedUrl,
-        tokenFetcher: TokenFetcher<OdspResourceTokenFetchOptions>,
-        throwOnNullToken: boolean,
-    ): (options: TokenFetchOptions, name: string) => Promise<string | null> {
-        return async (options: TokenFetchOptions, name: string) => {
-            // Telemetry note: if options.refresh is true, there is a potential perf issue:
-            // Host should optimize and provide non-expired tokens on all critical paths.
-            // Exceptions: race conditions around expiration, revoked tokens, host that does not care
-            // (fluid-fetcher)
-            return PerformanceEvent.timedExecAsync(
-                logger,
-                {
-                    eventName: `${name}_GetToken`,
-                    attempts: options.refresh ? 2 : 1,
-                    hasClaims: !!options.claims,
-                    hasTenantId: !!options.tenantId,
-                },
-                async (event) => tokenFetcher({
-                    ...options,
-                    siteUrl: resolvedUrl.siteUrl,
-                    driveId: resolvedUrl.driveId,
-                    itemId: resolvedUrl.itemId,
-                }).then((tokenResponse) => {
-                    const token = tokenFromResponse(tokenResponse);
-                    // This event alone generates so many events that is materially impacts cost of telemetry
-                    // Thus do not report end event when it comes back quickly.
-                    // Note that most of the hosts do not report if result is comming from cache or not,
-                    // so we can't rely on that here
-                    if (event.duration >= 32) {
-                        event.end({ fromCache: isTokenFromCache(tokenResponse), isNull: token === null });
-                    }
-                    if (token === null && throwOnNullToken) {
-                        throwOdspNetworkError(`${name} Token is null`, fetchTokenErrorCode);
-                    }
-                    return token;
-                }),
-                { cancel: "generic" });
-        };
-    }
+export function toInstrumentedOdspTokenFetcher(
```

Comment:
I believe this function should move into some small util ts file or its own file, as tree shaking will not shake all the in this file, and thus this new scenario will pick up all the factory related code

Reply:
@vladsud , I see how changing it from a private class function to a separate function makes a difference from a tree-shaking perspective, so that's great. And moving it to a separate file isn't a bad idea either. But FWIW, I don't think tree-shaking would care whether it's in the same file or not.

Reply:
Moved into new utils file.